### PR TITLE
docs(controllers): adds a section on controllers quorums and limitations

### PR DESCRIPTION
**Pull request by PaulRMellor**
**Documentation**

Adds a section to KRaft overview to describe controller quorums and current limitations with static controller quorums.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/assemblies/deploying/assembly-kraft-mode.adoc [hunk 1]
```diff
@@ -50,11 +50,42 @@ In the following example, three nodes perform a dual role and two nodes act only
...
+* Adding or removing controller roles from node pools
+* Switching node pools between dedicated controller roles and dual roles
+
+Static controller quorums also limit operations that require scaling. 
+For example, changing the storage type for a node pool with a controller role is not possible because it involves scaling the controller quorum. For non-JBOD storage, creating a new node pool with the desired storage type, adding it to the cluster, and removing the old one would require scaling operations, which are not supported. 
+In some cases, workarounds are possible.
+For instance, when modifying node pool roles to combine controller and broker functions, you can xref:proc-joining-node-pools-roles-str[add broker roles to controller nodes] instead of adding controller roles to broker nodes to avoid controller scaling. 
+However, this approach would require reassigning more data, temporarily affecting cluster performance.
+
+Migration between static and dynamic controller quorums is not currently supported, though it is expected to be introduced in a future Kafka release.
```
Comment (mimaison) [inline documentation/assemblies/deploying/assembly-kraft-mode.adoc:87]: Should we `Apache Kafka` instead of `Kafka`?

**Review by mimaison — APPROVED** [review]
LGTM, I just left a question

File: documentation/assemblies/deploying/assembly-kraft-mode.adoc [hunk 2]
```diff
@@ -50,11 +50,42 @@ In the following example, three nodes perform a dual role and two nodes act only
...
+* *Dynamic controller quorums* +
+This mode enables dynamic scaling of controllers without downtime. 
+New controllers join as observers, replicate the metadata log, and eventually become eligible to join the quorum.
+If a controller being removed is the active controller, it will step down from the quorum only after the new quorum is confirmed. 
+
+Scaling is useful not only for adding or removing controllers, but supports the following operations:
+
+* Replacing controllers because of hardware failure
+* Migrating clusters onto new hardware
+* Moving nodes from dedicated controller roles to dual roles or vice versa
```
Comment (scholzj) [inline documentation/assemblies/deploying/assembly-kraft-mode.adoc:68]: I don't think this really applies. On Kubernetes, you do the first two points by rescheduling the Pods. The 3rd point can also be done by adding or removing the broker roles.

On Kubernetes, the example of what might not be an obvious scaling operation is for example:
* Renaming a node pool (add a new node pool with new name and delete the old node pool with the old name)
* Changing non-JBOD storage (Create new node pool with new storage configuration and delete the old one)

The points you list here are more relevant to VMs / bare-metal Kafka.

File: documentation/assemblies/deploying/assembly-kraft-mode.adoc [hunk 3]
```diff
@@ -50,11 +50,42 @@ In the following example, three nodes perform a dual role and two nodes act only
...
+Dynamic controller quorums provide the flexibility to make these operations significantly easier to perform.
+
+== Limitations with static controller quorums
+
+Strimzi currently supports only static controller quorums.
+This limitation means dynamic scaling of controller quorums cannot be used to support the following:
+
+* Scaling up or down of node pools that have controller roles
+* Adding or removing controller roles from node pools
+* Switching node pools between dedicated controller roles and dual roles
```
Comment (scholzj) [inline documentation/assemblies/deploying/assembly-kraft-mode.adoc:79]: This is not about what Strimzi supports. It is about what Kafka does not support and should be formulated in that way. I would use something like this.

```suggestion
Migration between static and dynamic controller quorums is not currently supported by Apache Kafka, though it is expected to be introduced in a future release.
All pre-existing KRaft-based Apache Kafka clusters that use the static quorum have to keep using it.
In order to keep supporting existing KRaft-based clusters, Strimzi has to keep using the static quorum as well.

This limitation means dynamic scaling of controller quorums cannot be used to support the following:
* Adding or removing node pools with controller roles
* Adding the controller role to an existing node pool
* Removing the controller role from an existing node pool
* Scaling a node pool with the controller role
* Renaming a node pool with the controller role
```

File: documentation/assemblies/deploying/assembly-kraft-mode.adoc (hunk 1 above)
Comment (scholzj) [inline documentation/assemblies/deploying/assembly-kraft-mode.adoc:87]: This can be removed if the comment above is accepted.

```suggestion
```

**Comment by scholzj** [comment]
@PaulRMellor Did you forget to push the changes?

**Review by ppatierno — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Review by katheris — APPROVED** [review]

**Pull request merged**